packages/@aws-cdk/aws-cloudformation/lib/pipeline-actions.ts

```diff
@@ -481,3 +485,12 @@ interface StatementTemplate {
 }
 
 type StatementCondition = { [op: string]: { [attribute: string]: string } };
+
+function stackArnFromProps(props: PipelineCloudFormationActionProps): string {
```

Comment:
As mentioned in the previous CR, It's not a good practice to propagate in all the props here. It is confusing and has the potential to introduce unwanted data flow.

You only need `stackName` and `region`. Either just pass in two arguments `(stackName: string, region?: string)` or define an interface `PipelineCloudFormationDeploymentTarget` and have `PipelineCloudFormationActionProps` extend it. 

All the way through, just pass in that type and not the specific types of the props.

Reply:
Changed to using anonymous interfaces (IMO, extracting a customer-facing interface that is only used as an implementation detail seems like a bad idea).

Can you elaborate on your reasoning here? In my opinion, this new code is strictly worse than what I had there before. It's fragile (any change will have you changing multiple signatures), leaks implementation details of what do the functions do, and is less type safe (because `region` is optional, you can never pass it, and the type system will not complain). What is "unwanted data flow"?

Reply:
> This is not an implementation detail - those are public methods. 

Those are public methods of a non-exported class (`SingletonPolicy`). I understand that it's a Construct, but I don't think that fact alone automatically makes it public...?

> Inline interfaces won't translate well through jsii.

Inline interfaces were there before my change, so I assume they have to work.

Reply:
Oh, didn't realize these classes are private. Strike all that conversation :-) sorry